### ISSUE TITLE
Testing RA3 Support

### ISSUE THREAD

**Issue by danaues** [danaues/pylutron-caseta#7]
Hi Everyone
@MouseRatt, @JonGilmore, @daytonturner, @chrisdrackett, @mrmichaeldavis, @swamplynx

This is where we stand with official integration:
pylutron-caseta
https://github.com/gurumitts/pylutron-caseta/pull/101 was merged providing support for RA3 Processors
https://github.com/gurumitts/pylutron-caseta/pull/107 is merged, adding occupancy sensors for RA3
LED's have been implemented by cbw, and scenes are still a work in progress, but cbw's QSX code allows for button presses, essentially solving the scene problem once it's ready.

lutron-caseta HA integration:
https://github.com/home-assistant/core/pull/75323 is merged.  Required to support the RA3 non-serialized zones
https://github.com/home-assistant/core/pull/75324 is merged.  Fixing Sunnata Keypad Support

I think that once all the above PRs are merged, we'll have working RA3 support within the official release.

Is anyone available to test prior to official release?
I"m specifically interested in testing the occupancy sensors.  Does anyone have more than 1 sensor in an area?

There is a copy of my lutron_caseta HA integration here, we can install as a custom component for testing.
https://github.com/danaues/lutron_caseta

There are installation instructions in the README.  Unfortunately, we still have to manually install the custom pylutron-caseta module using pip inside the container.

Requires at least HA 8.7

Thanks.

**Comment by danaues** [comment]
Adding @iCSpotRun , @mrmichaeladavis

**Comment by peterxrock** [comment]
![113C44E3-E605-43C7-816B-30D2CEF4550A](https://user-images.githubusercontent.com/69395225/185508659-11416722-4d19-4bf5-b466-9429ce519f4a.jpeg)
Joining here. I have Sunnata 3b keypads, dimmers and one occupancy sensor. I have installed the custom component, but when trying to install pylutron, per the directions, the SSH console in HA doesn’t start with “ha”. I’m running on a VM, but I see Docker mentioned, so maybe that’s the issue.

**Comment by danaues** [comment]
@peterxrock, It won't work over SSH.  Try using the console screen on your VM,  Not the SSH connection.

**Comment by peterxrock** [comment]
I might be too novice for this. Hosting on a Synology VM so there's no built in console. I can SSH to the Synology but I don't know the commands to access the VM specifically. Sorry. 

Edit: One sec. Think I found it.

**Comment by peterxrock** [comment]
Install worked. When I try to add the component, I get this. ![image](https://user-images.githubusercontent.com/69395225/185515406-e3ba696a-78fe-47ec-8d45-e97bd60c2c3c.png)

**Comment by danaues** [comment]
looks like it's missing part of the message.  Try entering the IP address of your RA3 Processor and clicking submit

**Comment by danaues** [comment]
It should look like this:

![image](https://user-images.githubusercontent.com/24459240/185519025-5b3f8f63-027d-4d40-b1a8-c9c1504b5507.png)

**Comment by danaues** [comment]
Then this:
![image](https://user-images.githubusercontent.com/24459240/185519124-6c35ec4a-3aa6-4c0d-af86-f9fed8e0418b.png)

**Comment by peterxrock** [comment]
No luck. It just hung here for a few minutes.
<img width="423" alt="Screen Shot 2022-08-18 at 11 57 12 PM" src="https://user-images.githubusercontent.com/69395225/185545823-5d36e4ec-1df3-4225-9fc9-4483bd64eb77.png">

**Comment by peterxrock** [comment]
Realized I needed to push the button on the bridge. It paired. However, I have six keypads, but it hasn't recognized them all. Hasn't found any dimmers. Should I be listening for events?

**Comment by peterxrock** [comment]
Paired it successfully. I noticed it's picking up one room of the Triathlon shades, but none of the others. Let me know if you want to test anything specific!

**Comment by danaues** [comment]
@peterxrock 

To confirm, you're running at least HA 8.7.x ?

Can you add this to your configuration.yaml, restart and let me know if you see any lutron_caseta errors in the logs.

logger:
  default: info
  logs:
    custom_components.lutron_caseta: debug

**Comment by peterxrock** [comment]
I’m running 2022.8.6. I’m in the beta channel but I don't see 8.7. Will try to get it updated manually shortly.

**Comment by danaues** [comment]
@peterxrock I've just pushed some changes to the lutron-caseta custom component folder.

I don't have any shades to test with, but I think they were probably returning None serials, which created an error, and is why you only see the first one.  I've updated the component to handle None serialized zones for covers, switches and fans.  

update your config/custom_components/lutron_caseta with the latest from:
https://github.com/danaues/lutron_caseta

After updating, you should just be able to restart.  Let me know how it works out.

**Comment by peterxrock** [comment]
Thank you! All devices load now.

Keypads - Tested both Single Action and Toggle settings in Lutron Designer. I believe the component treats all button presses as Single Action. It would be ideal to have Toggle work, so you could use a single keypad button to toggle. I realize the automation logic can be "if light is on, then turn off/if off, then turn on". 

Shades - All of the covers load now; the single room of covers that loaded in the prior version show as "Restored" in HA but work. 

Naming - See attached. Many of the entities are picking up incorrect names.
<img width="298" alt="Screen Shot 2022-08-19 at 9 46 03 AM" src="https://user-images.githubusercontent.com/69395225/185649594-7e799f8d-4b2e-46ea-a27c-91e8338a54f7.png">

Logger added

Any recommendations on forcing 8.7 via the console on the VM? It's not showing through the UI Updates tab (even with beta channel turned on)

**Comment by mrmichaeladavis** [comment]
I will take a test on the weekend and report back!

**Comment by danaues** [comment]
@peterxrock Can you post your diagnostic.txt file for the integration:
Click on "Settings" > "Devices and Services" 
Select the 3 vertical dots icon at the bottom right of the Lutron Caseta integration box, and click "Download diagnostics" to get the file.

Also, can you provide example names that you were expecting for these:
![image](https://user-images.githubusercontent.com/24459240/185676700-d7a63b16-7209-45ca-a15e-40ae2cf921bc.png)

Thx

**Comment by peterxrock** [comment]
Attached. I ended up renaming the devices in Lutron, since it was pulling in as light.living_room_living_room_pendant. After a transfer and some reloads, it seems to pull the correct name now. I would note that when a device name is changed in Lutron, it creates a new device in HA and abandons the old one. This may be intended functionality. Just FYI.
[config_entry-lutron_caseta-548833c7f1dbe65ba9bf2c93245c28f4.json.txt](https://github.com/danaues/pylutron-caseta/files/9384435/config_entry-lutron_caseta-548833c7f1dbe65ba9bf2c93245c28f4.json.txt)

FYI - Everything is working really nicely. Love that I can single press the raise/lower buttons on the keypad to raise/lower the shades, instead of having to hold them down. I have it set up in Node-RED, too, with no issues.
<img width="520" alt="Screen Shot 2022-08-19 at 1 05 56 PM" src="https://user-images.githubusercontent.com/69395225/185680482-76746d52-2fca-4c6a-829a-4583e1f0e78b.png">

**Comment by peterxrock** [comment]
One issue - I have one room with multiple shade groups. One group is a single shade/door, one group for the two shades on either side, and one group for the single shade across the room. When using a keypad button to raise/lower, the shade groups are not in sync. Not by much, but enough that they're not aligned.

**Comment by peterxrock** [comment]
New issue - on Kitchen Keypad 1, the button 1 LED never turns off, even though it's set as Single Action in Lutron. It functions normally.

**Comment by peterxrock** [comment]
Logger: pylutron_caseta.smartbridge
Source: /usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py:485 
First occurred: 4:28:46 PM (40 occurrences) 
Last logged: 4:41:23 PM

Reconnecting...
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 460, in _monitor_once
    self._leap = await self._connect()
  File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 116, in _connect
    res = await open_connection(
  File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/leap.py", line 182, in open_connection
    reader, writer = await asyncio.open_connection(host, port, limit=limit, **kwds)
  File "/usr/local/lib/python3.10/asyncio/streams.py", line 47, in open_connection
    transport, _ = await loop.create_connection(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection
    raise exceptions[0]
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection
    sock = await self._connect_sock(
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock
    await self.sock_connect(sock, address)
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect
    return await fut
  File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.1.67', 8081)

**Comment by peterxrock** [comment]
Logger: homeassistant.helpers.integration_platform
Source: loader.py:674 
First occurred: August 19, 2022, 11:23:18 PM (3 occurrences) 
Last logged: August 19, 2022, 11:23:18 PM

Unexpected error importing lutron_caseta/recorder.py
Unexpected error importing lutron_caseta/media_source.py
Unexpected error importing lutron_caseta/system_health.py
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component
    platform = integration.get_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'

**Comment by danaues** [comment]
> New issue - on Kitchen Keypad 1, the button 1 LED never turns off, even though it's set as Single Action in Lutron. It functions normally.

Yes, LEDs are still on the To Do list.  Not supported yet.  I'll start looking into it once the occupancy support is merged

**Comment by danaues** [comment]
> 



> 

It kinda looks like the RA3 Processor was reset/restarted.  Did it come back on it own after this?

> Logger: pylutron_caseta.smartbridge Source: /usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py:485 First occurred: 4:28:46 PM (40 occurrences) Last logged: 4:41:23 PM
> 
> Reconnecting... Traceback (most recent call last): File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 460, in _monitor_once self._leap = await self._connect() File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 116, in _connect res = await open_connection( File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/leap.py", line 182, in open_connection reader, writer = await asyncio.open_connection(host, port, limit=limit, **kwds) File "/usr/local/lib/python3.10/asyncio/streams.py", line 47, in open_connection transport, _ = await loop.create_connection( File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection raise exceptions[0] File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection sock = await self._connect_sock( File "/usr/local/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock await self.sock_connect(sock, address) File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect return await fut File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb raise OSError(err, f'Connect call failed {address}') ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.1.67', 8081)

It kinda looks like the RA3 Processor was reset/restarted.  Did it come back on it own after this?

**Comment by danaues** [comment]
> Logger: homeassistant.helpers.integration_platform Source: loader.py:674 First occurred: August 19, 2022, 11:23:18 PM (3 occurrences) Last logged: August 19, 2022, 11:23:18 PM
> 
> Unexpected error importing lutron_caseta/recorder.py Unexpected error importing lutron_caseta/media_source.py Unexpected error importing lutron_caseta/system_health.py Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component platform = integration.get_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform cache[full_name] = self._import_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform return importlib.import_module(f"{self.pkg_path}.{platform_name}") File "/usr/local/lib/python3.10/importlib/**init**.py", line 126, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "", line 1050, in _gcd_import File "", line 1027, in _find_and_load File "", line 1004, in _find_and_load_unlocked ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'

I think you might have some extra files in your custom_components/lutron_caseta folder.

These files aren't supposed to be in there, and aren't in my repository.  Can you have a look?

lutron_caseta/recorder.py
lutron_caseta/media_source.py
lutron_caseta/system_health.py

**Comment by peterxrock** [comment]
> > 
> 
> 
> 
> 
> 
> 
> 
> > 
> 
> 
> 
> It kinda looks like the RA3 Processor was reset/restarted.  Did it come back on it own after this?
> 
> 
> 
> > Logger: pylutron_caseta.smartbridge Source: /usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py:485 First occurred: 4:28:46 PM (40 occurrences) Last logged: 4:41:23 PM
> 
> > 
> 
> > Reconnecting... Traceback (most recent call last): File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 460, in _monitor_once self._leap = await self._connect() File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 116, in _connect res = await open_connection( File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/leap.py", line 182, in open_connection reader, writer = await asyncio.open_connection(host, port, limit=limit, **kwds) File "/usr/local/lib/python3.10/asyncio/streams.py", line 47, in open_connection transport, _ = await loop.create_connection( File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection raise exceptions[0] File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection sock = await self._connect_sock( File "/usr/local/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock await self.sock_connect(sock, address) File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect return await fut File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb raise OSError(err, f'Connect call failed {address}') ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.1.67', 8081)
> 
> 
> 
> It kinda looks like the RA3 Processor was reset/restarted.  Did it come back on it own after this?

I wonder if this happens when there's a transfer? I was definitely transferring quite a bit yesterday.

**Comment by peterxrock** [comment]
> > Logger: homeassistant.helpers.integration_platform Source: loader.py:674 First occurred: August 19, 2022, 11:23:18 PM (3 occurrences) Last logged: August 19, 2022, 11:23:18 PM
> > Unexpected error importing lutron_caseta/recorder.py Unexpected error importing lutron_caseta/media_source.py Unexpected error importing lutron_caseta/system_health.py Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component platform = integration.get_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform cache[full_name] = self._import_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform return importlib.import_module(f"{self.pkg_path}.{platform_name}") File "/usr/local/lib/python3.10/importlib/**init**.py", line 126, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "", line 1050, in _gcd_import File "", line 1027, in _find_and_load File "", line 1004, in _find_and_load_unlocked ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'
> 
> I think you might have some extra files in your custom_components/lutron_caseta folder.
> 
> These files aren't supposed to be in there, and aren't in my repository. Can you have a look?
> 
> lutron_caseta/recorder.py lutron_caseta/media_source.py lutron_caseta/system_health.py

Nope, none of these files in the folder.

**Comment by danaues** [comment]
> > > 
> > 
> > 
> > > 
> > 
> > 
> > It kinda looks like the RA3 Processor was reset/restarted.  Did it come back on it own after this?
> > > Logger: pylutron_caseta.smartbridge Source: /usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py:485 First occurred: 4:28:46 PM (40 occurrences) Last logged: 4:41:23 PM
> > 
> > 
> > > 
> > 
> > 
> > > Reconnecting... Traceback (most recent call last): File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 460, in _monitor_once self._leap = await self._connect() File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 116, in _connect res = await open_connection( File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/leap.py", line 182, in open_connection reader, writer = await asyncio.open_connection(host, port, limit=limit, **kwds) File "/usr/local/lib/python3.10/asyncio/streams.py", line 47, in open_connection transport, _ = await loop.create_connection( File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection raise exceptions[0] File "/usr/local/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection sock = await self._connect_sock( File "/usr/local/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock await self.sock_connect(sock, address) File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect return await fut File "/usr/local/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb raise OSError(err, f'Connect call failed {address}') ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.1.67', 8081)
> > 
> > 
> > It kinda looks like the RA3 Processor was reset/restarted.  Did it come back on it own after this?
> 
> I wonder if this happens when there's a transfer? I was definitely transferring quite a bit yesterday.

Yes, happens to me on during a transfer too.  Mine comes back up on it's own after the processor is available again.

**Comment by danaues** [comment]
> > > Logger: homeassistant.helpers.integration_platform Source: loader.py:674 First occurred: August 19, 2022, 11:23:18 PM (3 occurrences) Last logged: August 19, 2022, 11:23:18 PM
> > > Unexpected error importing lutron_caseta/recorder.py Unexpected error importing lutron_caseta/media_source.py Unexpected error importing lutron_caseta/system_health.py Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component platform = integration.get_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform cache[full_name] = self._import_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform return importlib.import_module(f"{self.pkg_path}.{platform_name}") File "/usr/local/lib/python3.10/importlib/**init**.py", line 126, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "", line 1050, in _gcd_import File "", line 1027, in _find_and_load File "", line 1004, in _find_and_load_unlocked ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'
> > 
> > 
> > I think you might have some extra files in your custom_components/lutron_caseta folder.
> > These files aren't supposed to be in there, and aren't in my repository. Can you have a look?
> > lutron_caseta/recorder.py lutron_caseta/media_source.py lutron_caseta/system_health.py
> 
> Nope, none of these files in the folder.

might be related to running it on HA 8.6, maybe some things that were changed in 8.7 are causing this.  My clone of the lutron_caseta integration was from the 8.7 source tree.  Do you have to restart after you get this error?  Or does everything keep working?

**Comment by peterxrock** [comment]
> > > > Logger: homeassistant.helpers.integration_platform Source: loader.py:674 First occurred: August 19, 2022, 11:23:18 PM (3 occurrences) Last logged: August 19, 2022, 11:23:18 PM
> > > > Unexpected error importing lutron_caseta/recorder.py Unexpected error importing lutron_caseta/media_source.py Unexpected error importing lutron_caseta/system_health.py Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component platform = integration.get_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform cache[full_name] = self._import_platform(platform_name) File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform return importlib.import_module(f"{self.pkg_path}.{platform_name}") File "/usr/local/lib/python3.10/importlib/**init**.py", line 126, in import_module return _bootstrap._gcd_import(name[level:], package, level) File "", line 1050, in _gcd_import File "", line 1027, in _find_and_load File "", line 1004, in _find_and_load_unlocked ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'
> > > 
> > > 
> > > I think you might have some extra files in your custom_components/lutron_caseta folder.
> > > These files aren't supposed to be in there, and aren't in my repository. Can you have a look?
> > > lutron_caseta/recorder.py lutron_caseta/media_source.py lutron_caseta/system_health.py
> > 
> > 
> > Nope, none of these files in the folder.
> 
> might be related to running it on HA 8.6, maybe some things that were changed in 8.7 are causing this. My clone of the lutron_caseta integration was from the 8.7 source tree. Do you have to restart after you get this error? Or does everything keep working?

Everything keeps working. So far, none of the errors have stopped it. Is there an easy way to upgrade to 8.7 via the VM console? I tried some Googled commands but nothing happened.

**Comment by peterxrock** [comment]
Lots of these, but I believe due to 8.6, too:

Logger: pylutron_caseta.smartbridge
Source: runner.py:119 
First occurred: August 19, 2022, 11:23:18 PM (8 occurrences) 
Last logged: 3:45:23 PM

Occupancy group 1854 has a status but no sensors
Occupancy group 1866 has a status but no sensors

**Comment by danaues** [comment]
> Lots of these, but I believe due to 8.6, too:
> 
> Logger: pylutron_caseta.smartbridge Source: runner.py:119 First occurred: August 19, 2022, 11:23:18 PM (8 occurrences) Last logged: 3:45:23 PM
> 
> Occupancy group 1854 has a status but no sensors Occupancy group 1866 has a status but no sensors

These are expected.  The processor sends area status updates for areas that don't have any sensors.  The module just ignores them, and sends this warning.  Maybe I'll remove the warning, it isn't really important.

**Comment by danaues** [comment]
> > Lots of these, but I believe due to 8.6, too:
> > Logger: pylutron_caseta.smartbridge Source: runner.py:119 First occurred: August 19, 2022, 11:23:18 PM (8 occurrences) Last logged: 3:45:23 PM
> > Occupancy group 1854 has a status but no sensors Occupancy group 1866 has a status but no sensors
> 
> These are expected. The processor sends area status updates for areas that don't have any sensors. The module just ignores them, and sends this warning. Maybe I'll remove the warning, it isn't really important.

I've just updated this to send a debug message instead of a warning.  You can uninstall and re-install using pip to test.

**Comment by peterxrock** [comment]
> > > Lots of these, but I believe due to 8.6, too:
> > > Logger: pylutron_caseta.smartbridge Source: runner.py:119 First occurred: August 19, 2022, 11:23:18 PM (8 occurrences) Last logged: 3:45:23 PM
> > > Occupancy group 1854 has a status but no sensors Occupancy group 1866 has a status but no sensors
> > 
> > 
> > These are expected. The processor sends area status updates for areas that don't have any sensors. The module just ignores them, and sends this warning. Maybe I'll remove the warning, it isn't really important.
> 
> I've just updated this to send a debug message instead of a warning. You can uninstall and re-install using pip to test.

That worked!

**Comment by lsdave** [comment]
If you decide to integrate Lutron QSX i have a system that has lots of different device types that i could test with.

**Comment by peterxrock** [comment]
I noticed it's pulling in a device called "Bathroom Shower Fan". I do have a Bathroom Shower Fan that's currently unactivated, but it's the only unactivated switch showing up under Devices. When I open it, it has all the Lutron entities underneath it.
<img width="1038" alt="Screen Shot 2022-08-24 at 9 54 49 PM" src="https://user-images.githubusercontent.com/69395225/186563906-85e16730-e2f4-4321-869d-eae1f3f85023.png">
<img width="1159" alt="Screen Shot 2022-08-24 at 9 54 38 PM" src="https://user-images.githubusercontent.com/69395225/186563912-f72fae7f-bd87-49d4-98e5-19a287686c08.png">

**Comment by peterxrock** [comment]
I upgraded to 8.7 last night. It broke the integration (I assume this is because you sent one specific for 8.6). I downloaded the custom component in the first message of this thread, reinstalled, and now it won't let me add the custom component. Do I need to update something with pylutron?

**Comment by dcough** [comment]
> I upgraded to 8.7 last night. It broke the integration (I assume this is because you sent one specific for 8.6). I downloaded the custom component in the first message of this thread, reinstalled, and now it won't let me add the custom component. Do I need to update something with pylutron?

Also upgraded to 8.7 and now seeing a "Failed to Set Up" notice on the integration that was working a few days ago, and a notice to check the logs. 

The log shows:

> 2022-08-25 22:51:55.942 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry 192.168.1.70 RA3 for lutron_caseta
File "/config/custom_components/lutron_caseta/__init__.py", line 158, in async_setup_entry
File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 104, in connect
File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 543, in _login
File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 588, in _load_devices

Are you getting the same?

I'll attempt a re-install later this evening and report back.

**Comment by peterxrock** [comment]
> > I upgraded to 8.7 last night. It broke the integration (I assume this is because you sent one specific for 8.6). I downloaded the custom component in the first message of this thread, reinstalled, and now it won't let me add the custom component. Do I need to update something with pylutron?
> 
> 
> 
> Also upgraded to 8.7 and now seeing a "Failed to Set Up" notice on the integration that was working a few days ago, and a notice to check the logs. 
> 
> 
> 
> The log shows:
> 
> 
> 
> > 2022-08-25 22:51:55.942 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry 192.168.1.70 RA3 for lutron_caseta
> 
> File "/config/custom_components/lutron_caseta/__init__.py", line 158, in async_setup_entry
> 
> File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 104, in connect
> 
> File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 543, in _login
> 
> File "/usr/local/lib/python3.10/site-packages/pylutron_caseta/smartbridge.py", line 588, in _load_devices
> 
> 
> 
> Are you getting the same?
> 
> 
> 
> I'll attempt a re-install later this evening and report back. 

Yes, that's the same issue. I removed and reinstalled and it still didn't work. I ended up rolling back to a snapshot on 8.6.

**Comment by dcough** [comment]
> Yes, that's the same issue. I removed and reinstalled and it still didn't work. I ended up rolling back to a snapshot on 8.6.

Hm, I was able to re-install and no errors with launching anymore. My existing test automation was broken, but once I re-added the keypad and saved the automation it started working again.

I did the following to re-install (same as instructions but with a step of deleting the integration prior to the restart)
 
> enter the command "docker exec -it $(docker ps -f name=homeassistant -q) bash" and you get a "bash-5.1#" prompt
> enter the command "pip uninstall pylutron-caseta" and answer "y" when asked to proceed (or it will skip if not installed)
> enter the command "pip install git+https://github.com/danaues/pylutron-caseta@testing"
> Deleted my existing Lutron Caseta Integration using the HA web interface
> restarted home assistant using the HA web interface or client
> add the "Lutron Caseta" integration using the HA web interface or client

Going to try adding a few new devices to the network and we'll see how it goes.

**Comment by peterxrock** [comment]
> > Yes, that's the same issue. I removed and reinstalled and it still didn't work. I ended up rolling back to a snapshot on 8.6.
> 
> Hm, I was able to re-install and no errors with launching anymore. My existing test automation was broken, but once I re-added the keypad and saved the automation it started working again.
> 
> I did the following to re-install (same as instructions but with a step of deleting the integration prior to the restart)
> 
> > enter the command "docker exec -it $(docker ps -f name=homeassistant -q) bash" and you get a "bash-5.1#" prompt
> > enter the command "pip uninstall pylutron-caseta" and answer "y" when asked to proceed (or it will skip if not installed)
> > enter the command "pip install git+https://github.com/danaues/pylutron-caseta@testing"
> > Deleted my existing Lutron Caseta Integration using the HA web interface
> > restarted home assistant using the HA web interface or client
> > add the "Lutron Caseta" integration using the HA web interface or client
> 
> Going to try adding a few new devices to the network and we'll see how it goes.

Thanks! I was able to get it running again. Checking everything now. I still get this error in the log: 
Logger: homeassistant.helpers.integration_platform
Source: loader.py:674 
First occurred: 8:38:21 AM (3 occurrences) 
Last logged: 8:38:21 AM

Unexpected error importing lutron_caseta/recorder.py
Unexpected error importing lutron_caseta/system_health.py
Unexpected error importing lutron_caseta/media_source.py
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component
    platform = integration.get_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'

**Comment by dcough** [comment]
> Thanks! I was able to get it running again. Checking everything now. I still get this error in the log:

Happy to help, your install experience was a huge help to me! 

I'm not seeing errors, but I'm probably not looking in the right place. (Former front end web designer, minimal experience with anything developer related!) 

My config file has the following: 

> logger:
>   default: info
>   logs:
>     custom_components.lutron_caseta: debug
>     homeassistant.helpers.integration_platform: debug 

(I added the last one after seeing your error message).

And I'm looking in the "Full Logs" available from the HA UI in System > Logs. If that matches, then nothing being flagged on my system.

I've since added a few more RA3 devices and things seem to be running smoothly. 

@danaues as mentioned I'm pretty green with all of this stuff but let me know if there's any way I can be of help!

**Comment by peterxrock** [comment]
After upgrading to 8.7, I’m still not seeing the occupancy sensor working. It just stays on full time. ![image](https://user-images.githubusercontent.com/69395225/187086612-09a314bd-bf95-4ef3-bba7-cc567ee3387c.png)

**Comment by danaues** [comment]
> re

@dcough , thanks I appreciate the help testing!

**Comment by danaues** [comment]
@peterxrock is the sensor configured as occupancy or vacation within the lutron designer?  Can you send me another diagnostic.txt file?

Thanks!

**Comment by peterxrock** [comment]
> @peterxrock is the sensor configured as occupancy or vacation within the lutron designer? Can you send me another diagnostic.txt file?
> 
> Thanks!

Sorry. I realized the issue is the occupancy sensor isn't coming through at all - not that it's not updating. After 8.7, the automation was referencing a non-existent entity. Here are the logs.
<img width="573" alt="Screen Shot 2022-08-29 at 9 30 36 AM" src="https://user-images.githubusercontent.com/69395225/187226300-842c7b56-87b5-4dcd-b5c1-62adda3be2d7.png">
[config_entry-lutron_caseta-d8996d3cc3ff5fd1e256d94d0c397bcb.json.txt](https://github.com/danaues/pylutron-caseta/files/9445428/config_entry-lutron_caseta-d8996d3cc3ff5fd1e256d94d0c397bcb.json.txt)

**Comment by peterxrock** [comment]
Still getting this on 8.7, too:
Logger: homeassistant.helpers.integration_platform
Source: loader.py:674 
First occurred: 9:35:43 AM (3 occurrences) 
Last logged: 9:35:43 AM

Unexpected error importing lutron_caseta/recorder.py
Unexpected error importing lutron_caseta/media_source.py
Unexpected error importing lutron_caseta/system_health.py
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/integration_platform.py", line 40, in _async_process_single_integration_platform_component
    platform = integration.get_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 657, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 674, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'custom_components.lutron_caseta-main.recorder'

**Comment by danaues** [comment]
> ModuleNotFoundError: No module named 'custom_components.

@peterxrock , can you double check to make sure the folder name that you have the custom component in is

"/custom_components/lutron_caseta"  all in lower case.  The folder name has to match exactly the domain we've set in the manifest file.  

This forum post describes a similar problem that was fixed with capitalization of the folder name:
https://community.home-assistant.io/t/modulenotfounderror-custom-compontent/324670

**Comment by danaues** [comment]
> 

Did the sensor entity exist before the upgrade to 8.7?

**Comment by peterxrock** [comment]
> > ModuleNotFoundError: No module named 'custom_components.
> 
> @peterxrock , can you double check to make sure the folder name that you have the custom component in is
> 
> "/custom_components/lutron_caseta" all in lower case. The folder name has to match exactly the domain we've set in the manifest file.
> 
> This forum post describes a similar problem that was fixed with capitalization of the folder name: https://community.home-assistant.io/t/modulenotfounderror-custom-compontent/324670

As downloaded, it's "lutron_caseta-main". I changed it, and the issue resolved.

**Comment by peterxrock** [comment]
> Did the sensor entity exist before the upgrade to 8.7?

Yes, I think you dropped me an update to the component, and it was working for a bit on 8.6. Just confirming - I should see this in the entities list?

**Comment by danaues** [comment]
Yes, as a binary sensor

**Comment by peterxrock** [comment]
Got it. It's not coming through then.

**Comment by danaues** [comment]
@peterxrock maybe try re-installing through console with pip.  see if that fixes it.  Sounds like the occupancy support isn't part of the pylutron-caseta module that's in use.

**Comment by danaues** [comment]
make sure to uninstall first too, just the pip part

**Comment by peterxrock** [comment]
<img width="418" alt="Screen Shot 2022-08-29 at 11 07 11 AM" src="https://user-images.githubusercontent.com/69395225/187245230-c843da3a-b1f9-45a0-b49c-feb66e39a62d.png">

No luck. Just making sure I'm on the correct version.

**Comment by peterxrock** [comment]
Added a new keypad to Lutron this evening, but it's not showing up. I transferred a few times. 

There's also still the random Device (it changed from Bathroom Fan to Hallway Shades) that shows all the entities under it. Wonder if this is related.

[config_entry-lutron_caseta-d8996d3cc3ff5fd1e256d94d0c397bcb.json-2.txt](https://github.com/danaues/pylutron-caseta/files/9449103/config_entry-lutron_caseta-d8996d3cc3ff5fd1e256d94d0c397bcb.json-2.txt)

**Comment by dcough** [comment]
> Added a new keypad to Lutron this evening, but it's not showing up. I transferred a few times.
> 
> There's also still the random Device (it changed from Bathroom Fan to Hallway Shades) that shows all the entities under it. Wonder if this is related.

I installed an RRST-W4B-XX (SunnataKeypad) tonight along with a few dimmers and it's now showing in the integration as expected. Just to make sure, you activated the device between transfers, right? 

My install process was Physically Install, Add in Designer, Sync, Activate via phone, Sync back with designer, wait the 10+ min, Transfer, and then restart HA.

I also seem to have a "master device" with all the entities associated to it. In my case it's an RRST-8ANS switch that controls  a fan in our bedroom. Shows up as "None (Switched)" under the Device Tab in HA. This particular device was physically installed/added to Lutron after some of the entities to which it is now parenting.

Noticing that all my entities also have the same "Area" attached to them regardless of where they are in the house, which is also the same area as the Fan switch, which is not in the correct room. Wondering if this is a result of me not paying enough attention to available parameters in Designer or perhaps triggered by using Designer's "apply to all" functionality... I'll try to test both tomorrow.

**Comment by danaues** [comment]
@dcough, @peterxrock 
This switched device seems like it might be part of the problem, I'm going to try and re-create this on my setup.  
Thanks for the feedback, I'll let you know what I find.

**Comment by danaues** [comment]
@dcough, @peterxrock
I found a problem with how the HA integration was handling the 'None' serials from our zones.
I've pushed an update to the lutron_caseta custom component.

Try updating the custom component and I think it should fix this problem.

1. Delete the integration from HA "Settings>Devices and Services>Lutron Caseta Integration (3dots)>Delete"
2. Update the lutron_caseta custom component
3. Restart

Let me know how it goes.  Thx!

**Comment by peterxrock** [comment]
That fixed it for me! Thanks!

**Comment by JonGilmore** [comment]
@danaues thanks for all the work so far, this is awesome! I might be missing something, but is this expected to support scenes from 6 button hybrid keypads? I've got quite a few of them and while the actual local load control works, there aren't any scenes that get created.

**Comment by danaues** [comment]
@JonGilmore Unfortunately, scenes aren't supported yet, but it is on the to-do list.  

@cbw just added QSX support, and along with it:
Support for keypad LEDs (both providing current state, and allowing them to be turned on and off)
Ability to send a button tap for keypad buttons.

Both features will probably work with RA3 too.  So this might accomplish the same thing.  I haven't started testing it yet, bit will over the next few days.

https://github.com/gurumitts/pylutron-caseta/pull/106

**Comment by danaues** [comment]
> 

@peterxrock is the missing occupancy sensor back too?

**Comment by peterxrock** [comment]
> > 
> 
> 
> 
> @peterxrock is the missing occupancy sensor back too?

Good call. It is not. I’m going to try reactivating it in Lutron to double check.

**Comment by peterxrock** [comment]
@danaues Didn't seem to work. Adding my Lutron config here, in case anything obvious is missing. 

Like @JonGilmore said: this is awesome work. I was using an insane system of HomeKit input booleans to band-aid together the system - this makes it so much easier.

<img width="1041" alt="Screen Shot 2022-08-30 at 9 48 10 AM" src="https://user-images.githubusercontent.com/69395225/187468663-f3bd7c59-17bb-46af-9642-00d21394ee92.png">
<img width="1100" alt="Screen Shot 2022-08-30 at 9 47 50 AM" src="https://user-images.githubusercontent.com/69395225/187468665-613e9d77-9947-4ec1-a5ae-3bd89ead69ee.png">
<img width="622" alt="Screen Shot 2022-08-30 at 9 47 34 AM" src="https://user-images.githubusercontent.com/69395225/187468669-a229b29e-2492-4640-88d2-3eef28206b8a.png">

**Comment by JonGilmore** [comment]
I do also have a spare LRF2-OCR2B laying around that I'm not currently using (as I just moved into my home). I can activate it and test as well if it helps. Just lmk 😄

**Comment by danaues** [comment]
@JonGilmore If you don't mind, please add it and let me know if it works for you.  That would be a big help.  Thx

**Comment by iCSpotRun** [comment]
I'm happy to send an occupancy sensor to anyone for testing. I'm also still looking to do some tests with the Lutron RF modules for inclusion into the component.

I'm also happy to sponsor this work if someone wants to set that up here on the Repo.

Really appreciate all the work being done.

**Comment by danaues** [comment]
@dcough, @peterxrock, @JonGilmore 

I just pushed another update to the lutron_caseta custom component.  I'm simplifying the fix I made earlier, reducing the number of changes, and still accomplishing the same goal.  Makes it easier to get merged with the main HA/core tree.

Can you please update the custom component again and let me know if it's working for you.  You will have to delete the integration first, or it will duplicate most of your entities. 

    Delete the integration from HA "Settings>Devices and Services>Lutron Caseta Integration (3dots)>Delete"
    Update the lutron_caseta custom component
    Restart

Thanks for the help

**Comment by peterxrock** [comment]
@danaues Updated, but still no occupancy sensor.
[config_entry-lutron_caseta-8a5a8fe171936ca29adcaf305d0ae202.json.txt](https://github.com/danaues/pylutron-caseta/files/9454166/config_entry-lutron_caseta-8a5a8fe171936ca29adcaf305d0ae202.json.txt)

**Comment by danaues** [comment]
@peterxrock 
Ok thanks.  I wasn't expecting it to fix that issue yet, just want to make sure the entities are showing up under the correct device still, instead of all being listed under one random device.

For your occupancy sensor, can you send me the leap output of:
{"CommuniqueType": "ReadRequest", "Header": { "Url": "/device?where=IsThisDevice:false" }}

You can use the Lutron Designer.  Open the App, and open your project.
Click on the "Tools" from the top menu bar, and select "Terminal"
On the terminal window, wait for your processor IP to appear on the top line, then click "Connect"

Paste this:
{"CommuniqueType": "ReadRequest", "Header": { "Url": "/device?where=IsThisDevice:false" }}
into the bottom textbox and click "Send"

Send me a copy of the response.  I'm curious to see what DeviceType your processor is providing for your occupancy sensor.

Thanks.

**Comment by peterxrock** [comment]
8/30/2022 10:50:20 AM: LEAP connection established for a processor with IP address 192.168.1.67

8/30/2022 10:50:31 AM: Request Sent:- {"CommuniqueType": "ReadRequest", "Header": { "Url": "/device?where=IsThisDevice:false" }}
8/30/2022 10:50:31 AM: Response Received :-{"CommuniqueType":"ReadResponse","Header":{"MessageBodyType":"MultipleDeviceDefinition","StatusCode":"200 OK","Url":"/device?where=IsThisDevice:false"},"Body":{"Devices":[{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1854"},"AssociatedControlStation":{"href":"/controlstation/15368"},"href":"/device/1916","SerialNumber":71301233,"Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/1923"}],"LinkNodes":[{"href":"/device/1916/linknode/1918"}],"FirmwareImage":{"href":"/firmwareimage/1916"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Addressed"},{"Name":"Position 2","DeviceType":"SunnataSwitch","AssociatedArea":{"href":"/area/1878"},"AssociatedControlStation":{"href":"/controlstation/2195"},"href":"/device/2213","Parent":{"href":"/project"},"ModelNumber":"RRST-8ANS-XX","LocalZones":[{"href":"/zone/2220"}],"LinkNodes":[{"href":"/device/2213/linknode/2215"}],"FirmwareImage":{"href":"/firmwareimage/2213"},"DeviceClass":{"HexadecimalEncoding":"45f0101"},"AddressedState":"Unaddressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1902"},"AssociatedControlStation":{"href":"/controlstation/2666"},"href":"/device/2668","SerialNumber":71301326,"Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/2675"}],"LinkNodes":[{"href":"/device/2668/linknode/2670"}],"FirmwareImage":{"href":"/firmwareimage/2668"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/2285"},"href":"/device/15239","SerialNumber":79344606,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/15239/linknode/15240"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/2285"},"href":"/device/15255","SerialNumber":79344607,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/15255/linknode/15256"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1902"},"AssociatedControlStation":{"href":"/controlstation/2702"},"href":"/device/2704","SerialNumber":71301273,"Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/2711"}],"LinkNodes":[{"href":"/device/2704/linknode/2706"}],"FirmwareImage":{"href":"/firmwareimage/2704"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1866"},"AssociatedControlStation":{"href":"/controlstation/10498"},"href":"/device/7768","SerialNumber":71301238,"Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/7775"}],"LinkNodes":[{"href":"/device/7768/linknode/7770"}],"FirmwareImage":{"href":"/firmwareimage/7768"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1854"},"href":"/device/6248","SerialNumber":79337116,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6248/linknode/6249"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1854"},"href":"/device/6256","SerialNumber":79337167,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6256/linknode/6257"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1902"},"href":"/device/6386","SerialNumber":79337165,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6386/linknode/6387"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/4749"},"href":"/device/10734","SerialNumber":70191876,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/10734/linknode/10735"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1890"},"AssociatedControlStation":{"href":"/controlstation/7802"},"href":"/device/7804","SerialNumber":71124013,"Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/7811"}],"LinkNodes":[{"href":"/device/7804/linknode/7806"}],"FirmwareImage":{"href":"/firmwareimage/7804"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/4601"},"AssociatedControlStation":{"href":"/controlstation/15382"},"href":"/device/7822","Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/7829"}],"LinkNodes":[{"href":"/device/7822/linknode/7824"}],"FirmwareImage":{"href":"/firmwareimage/7822"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Unaddressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1902"},"AssociatedControlStation":{"href":"/controlstation/15412"},"href":"/device/15414","Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/15421"}],"LinkNodes":[{"href":"/device/15414/linknode/15416"}],"FirmwareImage":{"href":"/firmwareimage/15414"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Unaddressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1854"},"AssociatedControlStation":{"href":"/controlstation/15430"},"href":"/device/15432","Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/15439"}],"LinkNodes":[{"href":"/device/15432/linknode/15434"}],"FirmwareImage":{"href":"/firmwareimage/15432"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Unaddressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1902"},"href":"/device/6378","SerialNumber":70310183,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6378/linknode/6379"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/2285"},"href":"/device/7179","SerialNumber":70310224,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/7179/linknode/7180"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/2285"},"href":"/device/7187","SerialNumber":79337169,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/7187/linknode/7188"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/4749"},"href":"/device/10718","SerialNumber":70192135,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/10718/linknode/10719"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1902"},"href":"/device/15018","SerialNumber":70310221,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/15018/linknode/15019"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/2285"},"href":"/device/15247","SerialNumber":79344529,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/15247/linknode/15248"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataKeypad","AssociatedArea":{"href":"/area/1902"},"AssociatedControlStation":{"href":"/controlstation/8696"},"href":"/device/8698","SerialNumber":66421294,"Parent":{"href":"/project"},"ModelNumber":"RRST-W3RL-XX","LinkNodes":[{"href":"/device/8698/linknode/8700"}],"FirmwareImage":{"href":"/firmwareimage/8698"},"DeviceClass":{"HexadecimalEncoding":"1270101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataKeypad","AssociatedArea":{"href":"/area/4749"},"AssociatedControlStation":{"href":"/controlstation/10947"},"href":"/device/10949","SerialNumber":76657005,"Parent":{"href":"/project"},"ModelNumber":"RRST-W3RL-XX","LinkNodes":[{"href":"/device/10949/linknode/10951"}],"FirmwareImage":{"href":"/firmwareimage/10949"},"DeviceClass":{"HexadecimalEncoding":"1270101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1878"},"AssociatedControlStation":{"href":"/controlstation/2195"},"href":"/device/2197","SerialNumber":71301242,"Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/2204"}],"LinkNodes":[{"href":"/device/2197/linknode/2199"}],"FirmwareImage":{"href":"/firmwareimage/2197"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1866"},"href":"/device/6593","SerialNumber":79337113,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6593/linknode/6594"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1902"},"href":"/device/15003","SerialNumber":70310185,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/15003/linknode/15004"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataKeypad","AssociatedArea":{"href":"/area/1854"},"AssociatedControlStation":{"href":"/controlstation/1914"},"href":"/device/2950","SerialNumber":66421671,"Parent":{"href":"/project"},"ModelNumber":"RRST-W3RL-XX","LinkNodes":[{"href":"/device/2950/linknode/2952"}],"FirmwareImage":{"href":"/firmwareimage/2950"},"DeviceClass":{"HexadecimalEncoding":"1270101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataSwitch","AssociatedArea":{"href":"/area/2309"},"AssociatedControlStation":{"href":"/controlstation/4130"},"href":"/device/4132","Parent":{"href":"/project"},"ModelNumber":"RRST-8ANS-XX","LocalZones":[{"href":"/zone/4139"}],"LinkNodes":[{"href":"/device/4132/linknode/4134"}],"FirmwareImage":{"href":"/firmwareimage/4132"},"DeviceClass":{"HexadecimalEncoding":"45f0101"},"AddressedState":"Unaddressed"},{"Name":"Position 1","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1854"},"AssociatedControlStation":{"href":"/controlstation/7941"},"href":"/device/7943","SerialNumber":75337036,"Parent":{"href":"/project"},"ModelNumber":"L-REPPRO-BL","LinkNodes":[{"href":"/device/7943/linknode/7945"}],"DeviceClass":{"HexadecimalEncoding":"5040201"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1866"},"href":"/device/6301","SerialNumber":79337168,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6301/linknode/6302"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1878"},"href":"/device/6369","SerialNumber":70310186,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6369/linknode/6370"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataSwitch","AssociatedArea":{"href":"/area/2309"},"AssociatedControlStation":{"href":"/controlstation/4112"},"href":"/device/4114","Parent":{"href":"/project"},"ModelNumber":"RRST-8ANS-XX","LocalZones":[{"href":"/zone/4121"}],"LinkNodes":[{"href":"/device/4114/linknode/4116"}],"FirmwareImage":{"href":"/firmwareimage/4114"},"DeviceClass":{"HexadecimalEncoding":"45f0101"},"AddressedState":"Unaddressed"},{"Name":"Position 1","DeviceType":"SunnataKeypad","AssociatedArea":{"href":"/area/1866"},"AssociatedControlStation":{"href":"/controlstation/7714"},"href":"/device/7716","SerialNumber":76657007,"Parent":{"href":"/project"},"ModelNumber":"RRST-W3RL-XX","LinkNodes":[{"href":"/device/7716/linknode/7718"}],"FirmwareImage":{"href":"/firmwareimage/7716"},"DeviceClass":{"HexadecimalEncoding":"1270101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataDimmer","AssociatedArea":{"href":"/area/1866"},"AssociatedControlStation":{"href":"/controlstation/15394"},"href":"/device/15396","Parent":{"href":"/project"},"ModelNumber":"RRST-PRO-N-XX","LocalZones":[{"href":"/zone/15403"}],"LinkNodes":[{"href":"/device/15396/linknode/15398"}],"FirmwareImage":{"href":"/firmwareimage/15396"},"DeviceClass":{"HexadecimalEncoding":"45e0101"},"AddressedState":"Unaddressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/1890"},"href":"/device/6339","SerialNumber":79337166,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/6339/linknode/6340"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/2285"},"href":"/device/7195","SerialNumber":70310220,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/7195/linknode/7196"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Enclosure Device 001","DeviceType":"Unknown","AssociatedArea":{"href":"/area/4749"},"href":"/device/10726","SerialNumber":79344256,"Parent":{"href":"/project"},"LinkNodes":[{"href":"/device/10726/linknode/10727"}],"DeviceClass":{"HexadecimalEncoding":"30a0401"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataKeypad","AssociatedArea":{"href":"/area/2285"},"AssociatedControlStation":{"href":"/controlstation/3844"},"href":"/device/3982","SerialNumber":76657006,"Parent":{"href":"/project"},"ModelNumber":"RRST-W3RL-XX","LinkNodes":[{"href":"/device/3982/linknode/3984"}],"FirmwareImage":{"href":"/firmwareimage/3982"},"DeviceClass":{"HexadecimalEncoding":"1270101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"RPSCeilingMountedOccupancySensor","AssociatedArea":{"href":"/area/1866"},"AssociatedControlStation":{"href":"/controlstation/7021"},"href":"/device/7023","SerialNumber":56628547,"Parent":{"href":"/project"},"ModelNumber":"LRF2-OCR2B-P","LinkNodes":[{"href":"/device/7023/linknode/15377"}],"DeviceClass":{"HexadecimalEncoding":"6080101"},"AddressedState":"Addressed"},{"Name":"Position 1","DeviceType":"SunnataKeypad","AssociatedArea":{"href":"/area/4601"},"AssociatedControlStation":{"href":"/controlstation/6930"},"href":"/device/8730","SerialNumber":76657000,"Parent":{"href":"/project"},"ModelNumber":"RRST-W3RL-XX","LinkNodes":[{"href":"/device/8730/linknode/8732"}],"FirmwareImage":{"href":"/firmwareimage/8730"},"DeviceClass":{"HexadecimalEncoding":"1270101"},"AddressedState":"Addressed"}]}}

**Comment by danaues** [comment]
Great!  The DeviceType of your occupancy sensor is "RPSCeilingMountedOccupancySensor", which is different than "RPSOccupancySensor" that I've seen for all others.  I just have to add it to the list of DeviceTypes the occupancy sensors filter for.

I'll push an update shortly and we can test.

**Comment by danaues** [comment]
@peterxrock ok, I"ve pushed an update to the pylutron_caseta module.

Update the module (pip uninstall and re-install), and restart.  You shouldn't have to delete the integration.

enter the command "login", and you get a "#" prompt
enter the command "docker exec -it $(docker ps -f name=homeassistant -q) bash" and you get a "bash-5.1#" prompt
enter the command "pip uninstall pylutron-caseta" and answer "y" when asked to proceed (or it will skip if not installed)
enter the command "pip install git+https://github.com/danaues/pylutron-caseta@testing"
restart home assistant using the HA web interface or client

Let me know if your occupancy sensor shows up... and works.

**Comment by peterxrock** [comment]
@danaues That did it! Amazing work.

**Comment by danaues** [comment]
I just pushed another update to lutron_caseta to match the actual changes made in HA PR# 77553
https://github.com/home-assistant/core/pull/77553

Thanks @peterxrock for helping track down and fix this issue.

Everyone might get some device duplication after this update.  You'll have to delete the integration, and re-add to fix.

Thx!

**Comment by peterxrock** [comment]
I noticed the occupancy sensor stopped updating for about an hour this afternoon. I restarted HA and it recovered. Anything in the logs? 

I also noticed if the processor loses power, the integrations needs to be restarted for it to function in HA again. Is it possible to reattempt connection, so if we lose power while away, it will recover?

<img width="428" alt="Screen Shot 2022-08-31 at 3 48 57 PM" src="https://user-images.githubusercontent.com/69395225/187779653-c21f84d3-06f9-4686-af2b-888024c4e193.png">


[config_entry-lutron_caseta-127d00b6c1b96e2ff01295c2b33cf02f.json.txt](https://github.com/danaues/pylutron-caseta/files/9464938/config_entry-lutron_caseta-127d00b6c1b96e2ff01295c2b33cf02f.json.txt)

**Comment by danaues** [comment]
I'll do some testing with re-connections after a processor reboot.  
I think this should be working, but it's part of the original caseta code, nothing added for RA3/QSX.

**Comment by peterxrock** [comment]
Updated to 2022.9 without thinking. I can roll back, but before I do, any update for this version?

**Comment by danaues** [comment]
@peterxrock , No worries. I'll release a custom_component version compatible with 2022.9 shortly.  Just testing it now.

**Comment by danaues** [comment]
Ok it's ready.

Re-install pylutron-caseta using (remember to pip uninstall pylutron-caseta first):
"pip install git+https://github.com/danaues/pylutron-caseta@testing-ha-2022-9"
** it's a different branch than before, testing-ha-2022-9 vs testing

Then for the custom_component, use this branch that is specific for 2022-9:
https://github.com/danaues/lutron_caseta/tree/ha-2022-9

If you used git to install the component, you can just go into the config/custom_components/lutron_caseta folder and type
git pull
git switch ha-2022-9

Restart and you should be good!

**Comment by danaues** [comment]
LED support is now available for testing too.  Thanks to @cbw  Please let me know if it's working

**Comment by peterxrock** [comment]
> LED support is now available for testing too.  Thanks to @cbw  Please let me know if it's working
> 
> 
> 
> 

On it! For LED support, should we be using Single Scene or Toggle in Designer? The LED should just briefly illuminate when the button is pushed, but won't stay on, since it's not treated as a toggle in HA, right?

**Comment by danaues** [comment]
Shouldn’t matter.  For each button led, HA will expose a switch. switch.basement_bedroom_Position1_SunnataKeypad_Button1_LED. or something like that

you can query the status of the LED by checking the entity state.  You can switch the LED on or off by call the service switch.turn_on etc… just like a normal switch.

RA3 will automatically do it’s own thing in the background for lighting the LED, depending on how you have the scenes configured.  But if you leave a button un-programmed in RA3 designer, you can control the LED status from home assistant.  Like maybe you want to control a non-lutron light from a keypad button.

**Comment by danaues** [comment]
Next feature that’s coming is allowing you to push a keypad button from HA.  Essentially allowing you to initiate scenes from HA.  It’s a work in progress

**Comment by danaues** [comment]
This is the blueprint I created for managing the LED status based on another entity.

https://gist.github.com/danaues/996621fee59d7176283f5f47f068165c#file-lutron_led_status_handler-yaml

**Comment by danaues** [comment]
You should be able to click "import blueprint" from the blueprint page and paste the above link in

It watches for a state change on the entity you choose, and also sets the LED when HA starts, to try and catch any led sync issues from HA restarting.

**Comment by peterxrock** [comment]
Could the Blueprint track the status of switch entities? I have one keypad button that turns on a switch - would like to track that.

**Comment by danaues** [comment]
It should do that. Just select your switch entity from 2nd dropdown, and have it watch for 'on' in the 3rd dropdown.

**Comment by peterxrock** [comment]
> It should do that. Just select your switch entity from 2nd dropdown, and have it watch for 'on' in the 3rd dropdown.

Not showing any switch entities under the drop down. Just lights.

**Comment by danaues** [comment]
You're right! I have it filtering for lights.  
if you edit the .yaml file in config/blueprints/automations/danaues..... 
delete the line that days domain: light (I think it's line 18)

    track_entity:
      name: Entity to track state of
      description: Light or Group Entity you want the LED to match the state of
      default: ""
      selector:
        entity:
          domain: light

**Comment by danaues** [comment]
You might have to restart or rename the file for HA to pick up the changes

**Comment by daytonturner** [comment]
I'm trying the new lutron_caseta and pylutron-caseta, the previous branch i was on was ra3-occupancy, on 2022.7 i believe.

I noticed that all my devices and entities are now showing up twice, with the first one showing up as unavailable.

e.g. I have an Ensuite Under Cabinet light that shows up as light.ensuite_uc - after upgrading now i see it, as well as light.ensuite_uc_2.

Tried a quick disable/restart/enable/restart but that didnt help.  Do I need to fully remove the integration and re-add it?  I wonder why this happened in the first place though. Some internal naming change?

**Comment by danaues** [comment]
@daytonturner some stuff was changed that causes a rename of the entities.  Best way is to delete the integration, then re-add it.  Everything will only show up once.  Naming should be the same as original.

**Comment by danaues** [comment]
Delete the integration from HA "Settings>Devices and Services>Lutron Caseta Integration (3dots)>Delete"

**Comment by daytonturner** [comment]
OK Ive done that, and then tried re-adding the bridge the same way it had been added the first time, in configuration.yaml, like:

lutron_caseta:
  - host: 10.2.5.179
    keyfile: caseta.key
    certfile: caseta.crt
    ca_certs: caseta-bridge.crt

Upon restart, it tries to reconfigure and re-add the bridge but it fails, and log shows

"Timeout while trying to connect to bridge at 10.2.5.179"

Attempting to connect to that IP on port 8081 does in fact connect, so not sure what I'm overlooking here..

**Comment by daytonturner** [comment]
oooookay, well i just restarted for the third time and now it works fine.  So,.. nevermind! Maybe the socket was stuck in TIME_WAIT or something.  Thanks!

**Comment by peterxrock** [comment]
Keypad LEDs work well. Nice work. I'd remove the warnings from the logs. I wonder if these are because the keypad LEDs are briefly "on" when pressed (due to RA) or if this is truly an HA overlap. 
<img width="924" alt="Screen Shot 2022-09-14 at 8 59 17 AM" src="https://user-images.githubusercontent.com/69395225/190175123-75ae76a3-156c-4de9-ab39-2d3ceca05790.png">

**Comment by danaues** [comment]
Thanks!  Yes, easy to fix.  I'll upload an updated version of the automation this evening.

**Comment by danaues** [comment]
@peterxrock , check out this new version of the automation.

now allows tracking state of any entity, not just lights
fixed a bug in the state selection
changed automation type to Restart to eliminate the errors in the log

https://gist.github.com/danaues/996621fee59d7176283f5f47f068165c

**Comment by peterxrock** [comment]
Getting this one here and there:

Logger: pylutron_caseta.smartbridge
Source: runner.py:119 
First occurred: 1:28:29 PM (1 occurrences) 
Last logged: 1:28:29 PM

Occupancy group 1866 has sensors but no status

**Comment by peterxrock** [comment]
Possible to get this updated for 9.6? Have a few unrelated bugs so I'd like to upgrade.

**Comment by peterxrock** [comment]
Getting some new odd errors:  

Logger: homeassistant.helpers.service
Source: helpers/service.py:635 
First occurred: 12:39:19 PM **(5753 occurrences)** 
Last logged: 6:24:20 PM

Unable to find referenced entities switch.kitchen_keypad_position_1_sunnatakeypad_accent_lights_led or it is/they are currently not available
Unable to find referenced entities switch.kitchen_keypad_position_1_sunnatakeypad_first_floor_lights_led or it is/they are currently not available
Unable to find referenced entities switch.living_room_keypad_position_1_sunnatakeypad_accent_lights_led or it is/they are currently not available
Unable to find referenced entities switch.living_room_keypad_position_1_sunnatakeypad_first_floor_lights_led or it is/they are currently not available
Unable to find referenced entities switch.upstairs_hallway_keypad_position_1_sunnatakeypad_first_floor_lights_led or it is/they are currently not available

However, everything seems to work normally.

**Comment by peterxrock** [comment]
Added a bunch of occupancy sensors today. Same error messages, but they function just fine:

Logger: pylutron_caseta.smartbridge
Source: runner.py:119 
First occurred: 4:14:47 PM (5 occurrences) 
Last logged: 4:14:47 PM

Occupancy group 1866 has sensors but no status
Occupancy group 1902 has sensors but no status
Occupancy group 2309 has sensors but no status
Occupancy group 4601 has sensors but no status
Occupancy group 4749 has sensors but no status

**Comment by peterxrock** [comment]
Getting this after the recent processor firmware:

Logger: hass_nabucasa.iot
Source: runner.py:119 
First occurred: 1:37:03 PM (15 occurrences) 
Last logged: 2:40:13 PM

Connection closed: Closed by server. A newer instance connected (4002)

Also getting the "already running" on the LED blueprint still

**Comment by peterxrock** [comment]
Any updates for v10? I may have to abandon soon. Stuck on 9 with a bunch of script issues. Thanks!

**Comment by danaues** [comment]
Everything is merged into v11.  Can you upgrade tomorrow once it's released?  It has everything.

Alternatively, the dev branch of this repo has the everything too.  
https://github.com/danaues/lutron_caseta

Will work on v9 or v10 (or11).  The only issue with 9 was that there were changed made to HA that required alterations in all integrations.  This hasn't been the case for v10 or v11

**Comment by danaues** [comment]
You don't have to do anything involving shell, or docker anymore.  Just the custom_component folder is enough now that 
pylutron-caseta is fully updated and released (0.17.1)

Thanks for all your help testing!

**Comment by danaues** [comment]
Once you upgrade to HA 11.0 make sure to remove your custom_components/lutron_caseta folder.  It won't be necessary anymore, and will prevent you from receiving future updates to the integration.

Cheers!

**Comment by peterxrock** [comment]
Oh awesome. Super exciting. I’ll report back tomorrow!

**Comment by peterxrock** [comment]
*EDIT: Keypads are not functioning: 

<img width="1206" alt="Screen Shot 2022-11-03 at 1 25 34 PM" src="https://user-images.githubusercontent.com/69395225/199804387-8ed34bf0-03ee-4976-ae93-840ce4c04bf4.png">
[config_entry-lutron_caseta-4ce99760b8abb50e7f5c7e1380cb22d8.json.txt](https://github.com/danaues/pylutron-caseta/files/9932049/config_entry-lutron_caseta-4ce99760b8abb50e7f5c7e1380cb22d8.json.txt)

**Comment by danaues** [comment]
You just have to edit your keypad automations, and select the correct button from the trigger drop-down. 

The full version imports the button names instead of the generic “First button” or “Second button”.   

This is the consequence of being a beta tester,  but it’s a simple fix within your automations.

**Comment by peterxrock** [comment]
Got it. Thanks! How about this one?

<img width="641" alt="Screen Shot 2022-11-03 at 2 34 20 PM" src="https://user-images.githubusercontent.com/69395225/199817907-b581ec98-a4d5-4d2a-90da-b6d44ed3e7bf.png">

**Comment by danaues** [comment]
It’s just a quirk about how the processor reports occupancy of some zones that don’t have any occupancy sensors. You can safely ignore it.  

I’ll look into suppressing it (drop the reporting category down from warning to debug) in a future release.

**Comment by peterxrock** [comment]
Awesome. Amazing work on this. It really opens up the system. Thanks again!

**Comment by mrmichaeladavis** [comment]
Hey all, run testing today and all 89 of my switches and shades with RA3 proc worked flawlessly with HA 2022.11.1  - Thanks!

**Comment by clarencetso** [comment]
@danaues connected my RA3 processor system which includes Triathlon shades, Palladiom shades, Sunnata hybrid/non-hybrid keypads, Sunnata dimmers, Sunnata switches, and Pico remotes. Amazing work and very thankful for folks like you out there.
